modules/globebrowsing/src/geojson/geojsoncomponent.cpp

```diff
@@ -0,0 +1,743 @@
+        dictionary.hasKey(KeyDesc) ? dictionary.value<std::string>(KeyDesc) : ""
+    })
+    , _enabled(EnabledInfo, true)
+    , _globeNode(globe)
+    , _geoJsonFile(FileInfo)
+    , _heightOffset(HeightOffsetInfo, 0.f, -1e12f, 1e12f)
+    , _latLongOffset(
+        CoordinateOffsetInfo,
+        glm::vec2(0.f),
+        glm::vec2(-90.0),
+        glm::vec2(90.f)
+    )
+    , _drawWireframe(DrawWireframeInfo, false)
+    , _preventUpdatesFromHeightMap(PreventHeightUpdateInfo, false)
+    , _forceUpdateHeightData(ForceUpdateHeightDataInfo)
+    , _lightSourcePropertyOwner({ "LightSources", "Light Sources" })
+    , _featuresPropertyOwner({ "Features", "Features" })
+    , _pointRenderModeOption(
+        PointRenderModeInfo,
+        properties::OptionProperty::DisplayType::Dropdown
+    )
+    , _centerLatLong(
+        CentroidCoordinateInfo,
+        glm::vec2(0.f),
+        glm::vec2(-90.f, -180.f),
+        glm::vec2(90.f, 180.f)
+    )
+    , _flyToFeature(FlyToFeatureInfo)
+{
+    const Parameters p = codegen::bake<Parameters>(dictionary);
+
+    _enabled = p.enabled.value_or(_enabled);
+    addProperty(_enabled);
+
+    _opacity = p.opacity.value_or(_opacity);
+    addProperty(_opacity);
+    addProperty(_fade);
+
+    _geoJsonFile = p.file.string();
+    _geoJsonFile.setReadOnly(true);
+    addProperty(_geoJsonFile);
+
+    _ignoreHeightsFromFile = p.ignoreHeights.value_or(_ignoreHeightsFromFile);
+
+    const float minGlobeRadius = static_cast<float>(
+        _globeNode.ellipsoid().minimumRadius()
+    );
+
+    _heightOffset = p.heightOffset.value_or(_heightOffset);
+    _heightOffset.onChange([this]() { _heightOffsetIsDirty = true; });
+    _heightOffset.setMinValue(-0.9f * minGlobeRadius);
+    _heightOffset.setMaxValue(5.f * minGlobeRadius);
+    addProperty(_heightOffset);
+
+    _latLongOffset = p.coordinateOffset.value_or(_latLongOffset);
+    _latLongOffset.onChange([this]() { _dataIsDirty = true; });
+    addProperty(_latLongOffset);
+
+    if (p.defaultProperties.has_value()) {
+        _defaultProperties.createFromDictionary(*p.defaultProperties);
+    }
+    addPropertySubOwner(_defaultProperties);
+
+    _defaultProperties.pointTexture.onChange([&]() {
+        std::filesystem::path texturePath = _defaultProperties.pointTexture.value();
+        // Not ethat an empty texture is also valid => use default texture from module
+        if (std::filesystem::is_regular_file(texturePath) || texturePath.empty()) {
+            _textureIsDirty = true;
+        }
+        else {
+            LERROR(fmt::format(
+                "Provided texture file does not exist: '{}'",
+                _defaultProperties.pointTexture
+            ));
+        }
+    });
+
+    _defaultProperties.shouldTesselate.onChange([&]() { _dataIsDirty = true; });
+    _defaultProperties.tesselationLevel.onChange([&]() { _dataIsDirty = true; });
+    _defaultProperties.tesselationMaxDistance.onChange([&]() { _dataIsDirty = true; });
```

Comment:
No difference in actual code that is generated, but `&` automatically captions everything by reference, making it harder to see which variables are actually captured.  In particular `this` since we don't write it out.

Reply:
Thanks for the clarification 👍